python_modules/dagster/dagster/_core/pipes/client.py

```diff
@@ -170,6 +181,14 @@ def on_opened(self, opened_payload: PipesOpenedData) -> None:
         that can only be obtained from the external process.
         """
 
+    def on_launched(self, params: PipesLaunchedData) -> None:
+        """Can be called manually to submit extra information about the launched process
+        to the message reader.
+
+        By default this is a no-op. Specific message readers can override this to action information
+        that can only be obtained on the client side after the external process has been launched.
+        """
```

Comment:
I think this should just set `self._launched_payload` (like `on_opened`). We will let customization happen in other areas that inspect `launched_payload`. Let's do this for a docstring:

```
"""Hook that is called if `PipesSession.report_launched()` is called. Sets `launched_payload` on the message reader.

This hook is not necessarily called in every pipes session. It is useful primarily when we wish to condition the behavior of the message reader on some parameter that is only available after external process launch (such as a run id in the external system). The code calling `open_pipes_session()` is responsible for calling `PipesSession.report_launched()` if using a message reader that accesses `launched_payload`.
"""

Reply:
`PipesMessageReader.on_opened` currently doesn't do anything, it's also a no-op by default

Reply:
My bad you're right. In that case let's stay consistent. Just set `_launched_payload` on `PipesBlobStoreMessageReader` (which is where `_opened_payload` is set). In that case we also can't have `launched_payload` and `opened_payload` as attribute annotations on `PipesMessageReader` since they aren't common to all instances.

Reply:
ok, and in upstack I will move it to `PipesThreadedMessageReader`.

Reply:
ok, and in upstack I will move it to `PipesThreadedMessageReader`.